cli/cmd/io/read.hpp

```diff
@@ -100,6 +112,98 @@ class read_command : public command
 
                     store_ntks(reader);
                 }
+                if (is_set("fgl"))
+                {
+                    if (!topology.empty())
+                    {
+                        if (std::filesystem::exists(filename))
+                        {
+                            if (std::filesystem::is_regular_file(filename))
+                            {
```

Comment:
Maybe we can do all of these calls as a pre-processing before this giant block to reduce code duplication.